azext_edge/edge/providers/orchestration/resources/dataflows.py

```diff
@@ -438,3 +560,344 @@ def show(self, name: str, instance_name: str, resource_group_name: str) -> dict:
 
     def list(self, instance_name: str, resource_group_name: str) -> Iterable[dict]:
         return self.ops.list_by_resource_group(resource_group_name=resource_group_name, instance_name=instance_name)
+
+    def _get_endpoint_host(
+        self,
+        endpoint_type: DataflowEndpointType,
+        host: Optional[str] = None,
+        storage_account_name: Optional[str] = None,
+        eventhub_namespace: Optional[str] = None,
+        hostname: Optional[str] = None,
+        port: Optional[str] = None,
+        **_
+    ) -> str:
+        processed_host = ""
+        if endpoint_type in [
+            DataflowEndpointType.DATAEXPLORER.value,
+            DataflowEndpointType.FABRICREALTIME.value,
+        ]:
+            processed_host = host
+        elif endpoint_type == DataflowEndpointType.DATALAKESTORAGE.value:
+            processed_host = f"https://{storage_account_name}.blob.core.windows.net"
+        elif endpoint_type == DataflowEndpointType.FABRICONELAKE.value:
+            processed_host = "https://onelake.dfs.fabric.microsoft.com"
+        elif endpoint_type == DataflowEndpointType.EVENTHUB.value:
+            processed_host = f"{eventhub_namespace}.servicebus.windows.net:9093"
+        elif endpoint_type in [
+            DataflowEndpointType.CUSTOMKAFKA.value,
+            DataflowEndpointType.AIOLOCALMQTT.value,
+            DataflowEndpointType.EVENTGRID.value,
+            DataflowEndpointType.CUSTOMMQTT.value,
+        ]:
+            processed_host = f"{hostname}:{port}"
+
+        return processed_host
+
+    def _process_authentication_type(
+        self,
+        endpoint_type: DataflowEndpointType,
+        settings: dict,
+        authentication_type: Optional[str] = None,
+        **kwargs
+    ):
+        # TODO: reduce usage of kwargs
+        # No authentication method required for local storage
+        if endpoint_type == DataflowEndpointType.LOCALSTORAGE.value:
+            return
+
+        if authentication_type:
+            authentication_method = authentication_type
+        else:
+            # Identify authentication method using the provided kwargs
+            authentication_method = self._identify_authentication_method(
+                **kwargs
+            )
+
+        should_check_missing_params = False
+        original_auth_method = settings.get("authentication", {}).get("method")
+        if not original_auth_method:
+            should_check_missing_params = True
+        elif original_auth_method and original_auth_method != authentication_method:
+            should_check_missing_params = True
+            settings["authentication"] = {
+                "method": authentication_method,
+            }
+
+        # Check if authentication method is allowed for the given endpoint type
+        if authentication_method not in DATAFLOW_ENDPOINT_AUTHENTICATION_TYPE_MAP[endpoint_type]:
+            supported_auths = list(sorted(DATAFLOW_ENDPOINT_AUTHENTICATION_TYPE_MAP[endpoint_type]))
```

Comment:
Doesn't `sorted` return a list? Wondering if casting to list is necessary?

Reply:
checked only sorted is enough, thanks!